server/api/dev.js

```diff
@@ -0,0 +1,214 @@
+  const past = new Date(NOW.getTime() - fiveYears).toISOString();
+  const future = new Date(NOW.getTime() + fiveYears).toISOString();
+  const futurest = new Date(NOW.getTime() + fiveYears + 1).toISOString();
+
+  /*
+                s_start   s_end     v_start   v_end
+    prerelease  future    futurest  future    futurest
+    submission  pastest   futurest  future    futurest
+    review      pastest   past      future    futurest
+    voting      pastest   past      pastest   futurest
+    results     pastest   past      pastest   past
+  */
+
+  return {
+    id: 'dev',
+    name: status === 'reset' ? 'submission' : status,
+    date,
+    env_level: 'dev',
+    year_end: yearEnd,
+    winners_thread_id: null,
+    valid_reddit_id: false,
+    submission_start: isAfter('prerelease') ? pastest : future,
+    submission_end: isAfter('submission') ? past : futurest,
+    vote_start: (isAfter('review') ? pastest : future) + 2, // avoid vote before submission
+    vote_end: (isAfter('voting') ? past : futurest) + 2, // avoid vote before submission
+    subtext: 'Dev subtext',
+    local_voting: true,
+    prompt: 'This is the prompt for the development contest.',
+  };
+};
+
+exports.contest = async ({ body: { status }, username }, res) => {
+  try {
+    const isExisting = (await db.any("SELECT EXISTS (SELECT 1 FROM contests WHERE id='dev')"))[0].exists;
+
+    if (status === 'reset') {
+      /*
+        ------- CONTEST -------
+        Create a dev contest ONLY if it doesn't exit already
+        Reset should not change contest timestamps to avoid changing status
+      */
+      if (!isExisting) {
+        await db.insert('contests', [getDevContest(status)]);
+      }
+
+      /*
+        ------- DELETE OLD DATA -------
+        Ensures developer account is in dev database
+        Create a fake "dev" user
+      */
+      await db.del('votes', { contest_id: 'dev' });
+      await db.del('contest_entries', { contest_id: 'dev' });
+      await db.any("DELETE FROM entries WHERE id LIKE 'dev%';");
+      await db.del('contest_categories', { contest_id: 'dev' });
+
+      /*
+        ------- CONTEST_CATEGORIES -------
+        Create categories for dev contest
+      */
+      const categories = [{ contest_id: 'dev', category: 'cat1' }, { contest_id: 'dev', category: 'cat2' }];
+      await db.insert('contest_categories', categories);
+
+      /*
+        ------- USERS -------
+        Ensures developer account is in dev database
+        Create a fake "dev" user
+      */
+      const users = await db.select('SELECT username FROM users');
+      if (!users.some((user) => user.username === username)) {
+        await db.insert('users', [{ username, contest_reminders: true, moderator: true }]);
+        users.push({ username });
+      }
+      if (!users.some((user) => user.username === 'dev')) {
+        await db.insert('users', [{ username: 'dev', contest_reminders: true, moderator: false }]);
```

Comment:
FYI, only moderators are required to be in the database since the modified_by (for submission status) is the only field that refers to the users table. The regular username column is just a string column with no dependencies.

Reply:
Ah ok. Do you want me to remove it, or just keep it in in case the keys change in the future?

Reply:
I think it's fine to keep it, just something that I figured wasn't entirely intuitive about the database and wanted to point out.